source/ShopPanel.cpp

```diff
@@ -364,7 +364,8 @@ void ShopPanel::DrawButtons()
 	string BUY = IsAlreadyOwned() ? (playerShip ? "_Install" : "_Cargo") : "_Buy";
 	bigFont.Draw(BUY,
 		buyCenter - .5 * Point(bigFont.Width(BUY), bigFont.Height()),
-		CanBuy() ? hoverButton == 'b' ? hover : active : inactive);
+		CanBuy(IsAlreadyOwned()) ? hoverButton == (IsAlreadyOwned() ? 'i' : 'b')
+		? hover : active : inactive);
```

Comment:
This wasn't so bad before but now is rather difficult to follow.
Perhaps instead determine the colour beforehand:
```c++
	const Color &buyTextColor;
	if(!CanBuy(IsAlreadyOwned()))
		buyTextColor = inactive;
	else
	{
		const char hoverCompare;

		if(IsAlreadyOwned())
			hoverCompare = 'i';
		else
			hoverCompare = 'b';

		if(hoverButton == hoverCompare)
			buyTextColor = hover;
		else
			buyTextColor = active;
	}
```